tests/js/articles.editor.spec.js

```diff
@@ -0,0 +1,103 @@
+
+describe( 'Articles block', () => {
+  beforeAll( async () => {
+    // This helps in overriding Wordpress dialogs preventing actions.
+    await enablePageDialogAccept();
+  } );
+  beforeEach( async () => {
+    // Before running each test, go to the create post page.
+    await createNewPost({
+      postType: 'page',
+      title: 'Test Articles block',
+    } );
+
+    // Insert block by title.
+    await insertBlock( 'Articles' );
+    await page.waitForSelector( '.articles-title-container' );
+  } , 40000);
+
+  // This is the first test, tests starts with it().
+  it ( 'is inserted into the Editor', async () => {
+    // Check if block was inserted
+    expect( await page.$( '[data-type="planet4-blocks/articles"]' ) ).not.toBeNull();
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  } );
+
+  it ( 'should work with all editor inputs', async () => {
+    await selectBlockByName( 'planet4-blocks/articles' );
+
+    // Add richtext inputs.
+    await clearPreviousTextWithPlaceholder( 'Enter title' );
+    await typeInInputWithPlaceholderLabel( 'Enter title', 'News' );
+    await typeInInputWithPlaceholderLabel( 'Enter description', 'Latest news details' );
+
+    // Add loadmore button text in wysiwyg editor.
+    await clearPreviousTextWithLabel( 'Button Text' );
+    await typeInInputWithPlaceholderLabel( 'Enter text', 'More News' );
+
+    // Add Setting inputs
+    await openSidebarPanelWithTitle( 'Setting' );
+    await clearPreviousTextWithLabel( 'Button Text' );
+    await typeInInputWithLabel( 'Button Text', 'Read more' );
+    await typeInInputWithLabel( 'Button Link', 'https://www.greenpeace.org' );
+    await clickElementByText( 'label', 'Open in a new Tab' );
+    await clearPreviousTextWithLabel( 'Articles count' );
+    await typeInInputWithLabel( 'Articles count', '4' );
+    await typeInDropdownWithLabel( 'Select Tags', 'Climate' );
+    await typeInDropdownWithLabel( 'Post Types', 'Story' );
+    await clickElementByText( 'label', 'Ignore categories' );
+
+    expect( await getEditedPostContent() ).toMatchSnapshot();
+  }, 50000 );
+
+  it ( 'should show a warning if the URL is wrong', async () => {
+    await typeInInputWithLabel( 'Button Link', 'this is not a URL' );
+
+    // The warning component should appear.
+    await expect( page ).toMatchElement( '.edit-post-sidebar .input_error' );
+
+    // The warning text should appear.
+    await expect( page ).toMatch( ARTICLES_INVALID_URL_WARNING );
+  } );
+
+  it ( 'should use only the manually chosen posts if any are entered', async () => {
+    await openSidebarPanelWithTitle( 'Setting' );
+
+    await typeInDropdownWithLabel( 'Manual override', 'Duis posuere' );
+    await typeInDropdownWithLabel( 'Manual override', 'Mauris quis dictum magna' );
+    await typeInDropdownWithLabel( 'Manual override', 'Lorem ipsum dolor sit amet' );
+
+    // The only 3 manually override post should appear in articles block.
```

Comment:
There might be an issue here: now the articles in manual override are always returned ordered by date, regardless of the order that they are entered in the input (see [this ticket](https://jira.greenpeace.org/browse/PLANET-5462) for more info). So I think this part will now be failing, but I'm not sure how we can test this behaviour properly since we would need to know the dates of the articles... 🤔 Maybe we can just test that they are rendered, without checking the order?